packages/zowe-explorer/src/utils/ProfilesUtils.ts

```diff
@@ -209,18 +210,77 @@ export class FilterDescriptor implements vscode.QuickPickItem {
 }
 
 export class ProfilesUtils {
-    public static getProfileInfo(envTheia: boolean): imperative.ProfileInfo {
+    public static getCredentialManagerOverride(): imperative.ICredentialManagerNameMap | undefined {
+        ZoweLogger.trace("ProfilesUtils.getCredentialManagerOverride called.");
+        const knownCredentialManagers = imperative.CredentialManagerOverride.getKnownCredMgrs();
+        const credentialManager = knownCredentialManagers.find((knownCredentialManager) => {
+            try {
+                return vscode.extensions.getExtension(knownCredentialManager.credMgrZEName);
+            } catch (err) {
+                return false;
+            }
+        });
+        if (!credentialManager) {
+            return undefined;
+        }
+        return imperative.CredentialManagerOverride.getCredMgrInfoByDisplayName(credentialManager.credMgrDisplayName);
+    }
+
+    public static async activateCredentialManagerOverride(
+        credentialManagerExtension: vscode.Extension<any>
+    ): Promise<imperative.ICredentialManagerConstructor | undefined> {
+        try {
+            ZoweLogger.trace("ProfilesUtils.activateCredentialManagerOverride called.");
+            const exports = await credentialManagerExtension.activate();
+            if (credentialManagerExtension.isActive && exports) {
+                return credentialManagerExtension.exports as imperative.ICredentialManagerConstructor;
+            }
+            return undefined;
+        } catch (err) {
+            throw new Error(localize("activateCredentialManagerOverride.failedToActivate", "Custom credential manager failed to activate"));
+        }
+    }
+
+    public static async updateCredentialManagerSetting(setting: string): Promise<void> {
+        ZoweLogger.trace("ProfilesUtils.updateCredentialManagerSetting called.");
+        const settingEnabled: boolean = SettingsConfig.getDirectValue(globals.SETTINGS_SECURE_CREDENTIALS_ENABLED);
+        if (settingEnabled) {
+            await globals.setGlobalSecurityValue(setting);
+            imperative.CredentialManagerOverride.recordCredMgrInConfig(setting);
+        }
+    }
+
+    public static async getProfileInfo(envTheia: boolean): Promise<imperative.ProfileInfo> {
         ZoweLogger.trace("ProfilesUtils.getProfileInfo called.");
-        const mProfileInfo = new imperative.ProfileInfo("zowe", {
-            requireKeytar: () => getSecurityModules("keytar", envTheia),
+        const credentialManagerMap = ProfilesUtils.getCredentialManagerOverride();
+        const customCredentialManagerExtension = vscode.extensions.getExtension(credentialManagerMap?.credMgrZEName ?? "");
```

Comment:
Instead of calling `getExtension` with an empty string, should we do something like this?
`const customCredentialManagerExtension = credentialManagerMap?.credMgrZEName && vscode.extensions.getExtension(credentialManagerMap.credMgrZEName)`